react/getting_started_with_react/keys_in_react.md

```diff
@@ -0,0 +1,126 @@
+### Introduction
+
+In this lesson we will cover keys in React. Keys are special props for our components and we'll learn why they are used.
+
+### Lesson overview
+
+By the end of the lesson you should be able to answer the following:
+
+- What are keys and why does React need them?
+- What are good and bad examples of keys?
+
+### Why does React need keys?
```

Comment:
May not needed since we're linking to the React docs, but a "What are keys" section just to briefly explain them (plus an example that could help users understand them, like "Think of keys as this thing that may be familiar to you....")